src/shrub/v3/evg_task_group.py

```diff
@@ -30,10 +34,15 @@ class EvgTaskGroup(BaseModel):
     max_hosts: Optional[int] = None
     share_processes: Optional[bool] = None
     setup_group_can_fail_task: Optional[bool] = None
-    setup_group_timeout_secs: Optional[Union[int, str]] = None
+    setup_group_timeout_secs: Optional[int] = None
     setup_group: Optional[List[EvgCommand]] = None
+    teardown_group_can_fail_task: Optional[bool] = None
```

Comment:
Can we add this to the class doc as well? Also, out of curiousity, is there timeout_specs for teardown_group?

Reply:
> Can we add this to the class doc as well?

Apologies, `teardown_group_can_fail_task` is actually undocumented and invalid according to the EVG validator (which is a surprise to me). Filed [DEVPROD-12438](https://jira.mongodb.org/browse/DEVPROD-12032) and commented it out for now (similar approach to https://github.com/evergreen-ci/shrub.py/pull/40).

> Is there timeout_secs for teardown_group?

Yes, there is a `teardown_group_timeout_secs`. Added for completeness.

Reply:
[DEVPROD-12438](https://jira.mongodb.org/browse/DEVPROD-12438) clarified `teardown_group_can_fail_task` is deliberately unsupported due to implementation concerns and there are no plans to support it either. Removed the field entirely accordingly.